src/test/ui/lint/rfc-2383-lint-reason/fulfilled_expect.rs

```diff
@@ -0,0 +1,61 @@
+// check-pass
+
+#![feature(lint_reasons)]
+#![warn(unused_variables)]
+
+#[expect(unused_variables)]
+fn check_specific_lint() {
+    let x = 2;
+}
+
+#[expect(unused)]
+fn check_lint_group() {
+    let x = 15;
+}
+
+#[expect(unused_variables)]
+fn check_multiple_lint_emissions() {
+    let r = 1;
+    let u = 8;
+    let s = 2;
+    let t = 9;
+}
+
+mod check_multiple_expected_lints {
+    #[expect(unused_variables, unused_mut)]
+    pub fn check_lint_1() {
+        // unused_variables should fulfill the expectation
+        let c = 17;
+    }
```

Comment:
This should trigger the unfulfilled lint for `unused_mut`. The reason it doesn't is that `unused` is globally allowed in compiletest. If you add a `#![warn(unused_mut`)]` at the top of the file, this should fail.

Maybe add a general test case where `expect` overrides `allow`/`warn` similar to the above tests where it's the other way around.

Reply:
> This should trigger the unfulfilled lint for `unused_mut`. The reason it doesn't is that `unused` is globally allowed in compiletest. If you add a `#![warn(unused_mut`)]` at the top of the file, this should fail.

The RFC document doesn't seem to define what happens if multiple lints have been listed, so it's probably open for discussion. This implementation sees the `expect` attribute as one expectation that can be fulfilled by any of the listed lints (This is also what the test should illustrate). I found this reasonable for an example like this:
```rs
#[expect(unused_mut, unused_value, reason = "This is work in progress, these are expected to happen")]
```
Here the reason stands for all of them and I don't care which of them is triggered. If I would like to be more precise, I can simply add more expect attributes. This gives the user more freedom IMO.

Reply:
Hm, that would mean, if I expect multiple lints, I would have to add multiple `expect` attributes. This seems really counter-intuitive to me. `allow(A, B)` allows the lints `A` and `B`, so `expect(A, B)` should also expect `A` and `B` and not `A` _or_ `B`.

Reply:
Yeah, I can see the reasoning behind this. I'll give it some more thought, but probably change the implementation. The change can be done by giving each listed lint it's on `LintExpectationId`. That again touch on the question of how that ID should be generated. I'll ask the incremental compilation Zulip stream for some thoughts on the ID before applying this change :upside_down_face:.

Thank you for the feedback!

Reply:
> If you add a `#![warn(unused_mut)]` at the top of the file, this should fail.

I've now also added the corresponding `warn` attributes for the lints in question to all test files :upside_down_face:

Reply:
The latest update now tracks each lint individually. :upside_down_face: